frame/contracts-registry/src/lib.rs

```diff
@@ -0,0 +1,271 @@
+	///
+	/// TWOX-NOTE: Safe, as increasing integer keys are safe.
+	#[pallet::storage]
+	#[pallet::getter(fn contracts)]
+	pub type Contracts<T: Config> = StorageMap<
+		_,
+		Twox64Concat,
+		ContractIndex,
+		ContractsData<T::MaxContractContentLen, T::MaxContractParties, T::AccountId>,
+		OptionQuery,
+	>;
+
+	/// Vec of parties signatures for each contract
+	///
+	/// TWOX-NOTE: Safe, as increasing integer keys are safe.
+	#[pallet::storage]
+	#[pallet::getter(fn parties_signatures)]
+	pub type PartiesSignatures<T: Config> = StorageMap<
+		_,
+		Twox64Concat,
+		ContractIndex,
+		BoundedVec<T::AccountId, T::MaxSignatures>,
+		OptionQuery,
+	>;
+
+	/// Vec of judge signatures for each contract
+	///
+	/// TWOX-NOTE: Safe, as increasing integer keys are safe.
+	#[pallet::storage]
+	#[pallet::getter(fn judge_signatures)]
+	pub(super) type JudgeSignatures<T: Config> = StorageMap<
+		_,
+		Twox64Concat,
+		ContractIndex,
+		BoundedVec<T::AccountId, T::MaxSignatures>,
+		OptionQuery,
+	>;
+
+	/// List of judges
+	#[pallet::storage]
+	#[pallet::getter(fn judges)]
+	pub type Judges<T: Config> =
+		StorageValue<_, BoundedVec<T::AccountId, T::MaxJudges>, ValueQuery>;
+
+	/// The number of contracts that have been made so far.
+	#[pallet::storage]
+	#[pallet::getter(fn contracts_count)]
+	pub type ContractsCount<T> = StorageValue<_, ContractIndex, ValueQuery>;
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		/// Add judge
+		///
+		/// The dispatch origin of this call must be _Signed_ and the sender must
+		/// have funds to cover the deposit.
+		///
+		/// - `judge`: Account that will be added as one of a judges
+		///
+		/// Emits `AddedJudge`.
+		#[pallet::call_index(0)]
+		#[pallet::weight(100_000)]
+		pub fn add_judge(origin: OriginFor<T>, judge: T::AccountId) -> DispatchResult {
+			let _ = T::AddJudgeOrigin::ensure_origin(origin)?;
+
+			let real_judges_count = Judges::<T>::decode_len().unwrap_or(0) as u32;
+			let max_judges = T::MaxJudges::get();
+			ensure!(real_judges_count < max_judges, Error::<T>::TooMany);
+
+			Judges::<T>::try_append(&judge).map_err(|_| Error::<T>::TooMany)?;
+
+			Self::deposit_event(Event::<T>::AddedJudge {
+				judge_id: real_judges_count + 1,
+				deposit: judge,
+			});
+			Ok(())
+		}
+
+		/// Judge can sign contract
+		///
+		/// The dispatch origin of this call must be _Signed_ and the sender must
+		/// have funds to cover the deposit.
+		///
+		/// - `contract_id`: The `id` of a contract that should be signed
+		///
+		/// Emits `JudgeSigned`.
+		#[pallet::call_index(1)]
+		#[pallet::weight(100_000)]
+		pub fn judge_sign_contract(
+			origin: OriginFor<T>,
+			contract_id: ContractIndex,
+		) -> DispatchResult {
+			let who = T::SubmitOrigin::ensure_origin(origin)?;
+			let is_judge = Judges::<T>::get().contains(&who);
+			ensure!(is_judge, Error::<T>::NotJudge);
+
+			JudgeSignatures::<T>::try_mutate(contract_id, |_signatures: &mut Option<BoundedVec<T::AccountId, T::MaxSignatures>>| -> DispatchResult {
```

Comment:
By convention `_` prefix is used for marking unused variables. Here `_signatures` is very much used.